No enumerate check in __iter__

bengineer19 opened the pull request:
## Steps

- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description
Fixes #2477 by ignoring the `enumerate` suggestion when defining `__iter__` and using `range(len(self))`.

## Type of Changes
<!-- Leave the corresponding lines for the applicable type of change: -->
|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |

## Related Issue
Closes #2477

PCManticore commented on pylint/checkers/refactoring.py:
```diff
@@ -1078,6 +1078,9 @@ def visit_for(self, node):
         if not isinstance(iterating_object, astroid.Name):
             return
+        # If we're defining __iter__ on self, enumerate won't work
+        if iterating_object.name == "self" and node.parent.name == "__iter__":
```
You'll need to make sure that `parent` is a scope though:
```
scope = node.scope()
if scope.name == '__iter__':
    ...
```

bengineer19 commented:
I've added a few test cases and updated the Changelog. I also switched to using `node.scope()` - could you check that I caught your drift correctly on that one? Thanks

PCManticore commented:
Thank you so much @bengineer19 !
